app/services/tracking.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * @module timed
+ * @submodule timed-services
+ * @public
+ */
+import Service  from 'ember-service'
+import service  from 'ember-service/inject'
+import moment   from 'moment'
+import computed from 'ember-computed-decorators'
+
+import {
+  task,
+  timeout
+} from 'ember-concurrency'
+
+/**
+ * Tracking service
+ *
+ * This contains some methods, the application needs on multiple routes
+ *
+ * @class TrackingService
+ * @extends Ember.Service
+ * @public
+ */
+export default Service.extend({
+  /**
+   * The store service
+   *
+   * @property {Ember.Store} store
+   * @public
+   */
+  store: service('store'),
+
+  /**
+   * The notify service
+   *
+   * @property {EmberNotify.NotifyService} notify
+   * @public
+   */
+  notify: service('notify'),
+
+  /**
+   * Init hook, get the current activity
+   *
+   * @method init
+   * @public
+   */
+  async init() {
+    this._super()
+
+    let actives = await this.get('store').query('activity', {
+      include: 'blocks,task,task.project,task.project.customer',
+      active: true
+    })
+
+    this.set('activity', actives.getWithDefault('firstObject', null))
+  },
+
+  /**
+   * Filter all customers
+   *
+   * @property {EmberConcurrency.Task} filterCustomers
+   * @public
+   */
+  filterCustomers: task(function* () {
+    yield timeout(500)
+
+    let customers = yield this.get('store').query('customer', {})
+
+    return customers
+  }).restartable(),
+
+  /**
+   * Filter all projects by customer
+   *
+   * @property {EmberConcurrency.Task} filterProjects
+   * @public
+   */
+  filterProjects: task(function* (customer) {
+    if (!customer) {
+      throw new Error('No customer selected')
+    }
+
+    yield timeout(500)
+
+    let projects = yield this.get('store').query('project', { customer })
+
+    return projects
+  }).restartable(),
+
+  /**
+   * Filter all tasks by project
+   *
+   * @property {EmberConcurrency.Task} filterTask
+   * @public
+   */
+  filterTasks: task(function* (project) {
+    if (!project) {
+      throw new Error('No project selected')
+    }
+
+    yield timeout(500)
+
+    let tasks = yield this.get('store').query('task', { project })
+
+    return tasks
+  }).restartable(),
```

Comment:
I wouldn't do that.. Just because there are 3 similar functions which filter different models, we don't have to put that into one function. Every filter is custom to the model so why put effort into that?